frame/salary/src/lib.rs

```diff
@@ -0,0 +1,245 @@
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Make periodic payment to members of a ranked collective according to rank.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+#![recursion_limit = "128"]
+
+use codec::{Decode, Encode, FullCodec, MaxEncodedLen};
+use scale_info::TypeInfo;
+use sp_arithmetic::traits::{Saturating, Zero};
+use sp_runtime::traits::{AtLeast32BitUnsigned, CheckedSub, Convert};
+use sp_std::{fmt::Debug, marker::PhantomData, prelude::*};
+
+use frame_support::{
+	dispatch::DispatchResultWithPostInfo, ensure, traits::RankedMembers, RuntimeDebug,
+};
+
+#[cfg(test)]
+mod tests;
+
+#[cfg(feature = "runtime-benchmarks")]
+mod benchmarking;
+pub mod weights;
+
+pub use pallet::*;
+pub use weights::WeightInfo;
+
+/// Payroll cycle.
+pub type Cycle = u32;
+
+// Can be implemented by Pot pallet with a fixed Currency impl, but can also be implemented with
+// XCM/MultiAsset and made generic over assets.
+pub trait Pay {
+	/// The type by which we measure units of the currency in which we make payments.
+	type Balance: AtLeast32BitUnsigned + FullCodec + MaxEncodedLen + TypeInfo + Debug;
+	/// The type by which we identify the individuals to whom a payment may be made.
+	type AccountId;
+	/// An identifier given to an individual payment.
+	type Id: FullCodec + MaxEncodedLen + TypeInfo + Clone + Eq + PartialEq + Debug;
+	/// The amount of currency with which we have to make payments in this period. It may be a fixed
+	/// value or reduce as calls to `pay` are made. It should be called once prior to the series of
+	/// payments to determine the overall budget and then not again until the next series of
+	/// payments are to be made.
+	fn budget() -> Self::Balance;
+	/// Make a payment and return an identifier for later evaluation of success in some off-chain
+	/// mechanism (likely an event, but possibly not on this chain).
+	fn pay(who: &Self::AccountId, amount: Self::Balance) -> Result<Self::Id, ()>;
+}
+
+#[derive(Encode, Decode, Eq, PartialEq, Clone, TypeInfo, MaxEncodedLen, RuntimeDebug)]
+pub struct StatusType<CycleIndex, BlockNumber, Balance> {
+	cycle_index: CycleIndex,
+	cycle_start: BlockNumber,
+	remaining_budget: Balance,
+}
+
+#[frame_support::pallet]
+pub mod pallet {
+	use super::*;
+	use frame_support::{dispatch::Pays, pallet_prelude::*};
+	use frame_system::pallet_prelude::*;
+
+	#[pallet::pallet]
+	#[pallet::generate_store(pub(super) trait Store)]
+	pub struct Pallet<T, I = ()>(PhantomData<(T, I)>);
```

Comment:
I think this provides a nice cut to start a `pallets` repo in the fellowship collective instead of adding it to paritytech/substrate.